### ISSUE TITLE
from Human to Plastic Puppet

### ISSUE THREAD

**Issue by ljiqy** [rinongal/StyleGAN-nada#5]
Thanks for the excellent work!
I want to train the generator from Human to Plastic Puppet using your adaptive layer-freezing approach, but I can't get the result(Figure 13 in the paper) from your train script, I change the improve_shape=True and lambda_global=1.0 in your script, but I'm not sure how to set the other parameters, such as "lambda_global"、"lambda_patch"、"lambda_texture"、"lambda_manifold" and so on.
can you share your train script for adaptive layer-freezing approach?

**Comment by rinongal** [comment]
Thank you for your interest in our work :)

The figures in the paper all use lambda_direction = 1.0 with all other lambdas set to 0 (with the exception of the 'Global loss' column in Fig 13 which uses lambda_global = 1.0 and lambda_direction = 0.0). 

I see that we unfortunately omitted the parameters used for the ablation figure. I'll add them in the next revision. The plastic puppets were generated with the following command:

```python train.py --size 1024 --batch 2 --n_sample 4 --output_dir ~/outputs/path --lr 0.002 --frozen_gen_ckpt /path/to/stylegan2_models/ffhq.pt --iter 601 --source_class "human" --target_class "plastic puppet" --lambda_direction 1.0 --auto_layer_k 5 --auto_layer_iters 1 --auto_layer_batch 8 --output_interval 50 --clip_models "ViT-B/32" --clip_model_weights 1.0 --mixing 0.0 --save_interval 100```

If you're using both CLIP ViT models (ViT-B/32 + ViT-B/16) that's probably where the main difference is coming from.

I uploaded the model checkpoint (and a few others used in the paper / project page / tweets) to [this google drive folder.](https://drive.google.com/drive/folders/1Z76nD8pXIL2O5f6xV8VjM4DUCmhbzn0l?usp=sharing)

**Comment by ljiqy** [comment]
Thank you for your detailed answer！
I got the correct result with your script and model checkpoint, but when I try to train generator from human to Avatar,I can't got good result, can you share you share your experience on how to adjust hyper-parameter?

**Comment by rinongal** [comment]
Happy to help!

For hyper-parameters, there are basically four knobs you can adjust that have meaningful impact:

1) Most importantly, try to make your text as unambiguous as possible. If you're aiming for Na'vi from the Avatar movie (the blue aliens), don't use "Avatar" as the target text since it can have different interpretations (for example blue arrow on head like Nickelodeon's Avatar). You can try "Na'vi" or "Avatar movie Na'vi" or even specific characters like "Neytiri from Avatar".
Another trick to try is to explicitly break realism. "human photo" to "avatar painting" is often easier than "human" to "avatar".

2) Try to change the CLIP models you use. From my experience, ViT-B/32 is better for global changes, ViT-B/16 is better for local ones. In most cases I'd try either both together, or just ViT-B/32.

3) You can try to enable style mixing (i.e. set --mixing=0.9). This is rarely helpful, but it can help with some big image changes.

4) You can play with the automatic layer parameters. I actually usually start without it (i.e. set auto_layer_k and auto_layer_iters to 0) since things train much faster that way, and often you get good results even without freezing. If you can't get things to work without it, just set auto_layers_k somewhere around 10. If the image still breaks apart early, lower it even more (3-5~ unless you're aiming for animal species changes). It's not very sensitive to exact values.

With all that said, not all changes will work. Face colors in particular are a bit of a hit-and-miss. Darth Maul and Kratos will work, but I didn't have any luck with Nebula, Gamora or Thanos. It's possible that Avatar will just not work with this model.

**Comment by ljiqy** [comment]
Thank you for sharing your valuable experience, which has been very helpful to me！

**Comment by rinongal** [comment]
Happy to help. I'm guessing that's it for now so I'm closing the issue, but if you have any other questions feel free to re-open it or make a new one.
